File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let createdMap = new Map()

    for (let i = 0; i < nums.length; i++) {
        let newDiff = target - nums[i]

        if (createdMap.has(newDiff)) {
            return [i, createdMap.get(newDiff)]
        }
        createdMap.set(nums[i], i)
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        pre_map = {}
        
        for i, n in enumerate(nums):
            difference = target - n
            if difference in pre_map:
                return [pre_map[difference], i]
            pre_map[n] = i
        return